models/google_ads__click_performance.sql

```diff
@@ -13,16 +13,16 @@ with base as (
         ad_group_id,
         criteria_id,
         gclid,
-        row_number() over (partition by gclid order by date_day) as rn
+        row_number() over (partition by gclid order by date_day) as rn,
```

Comment:
Could there be duplicate `gclid` across the sources? Trying to understand if we would also need to add `source_relation` to the partition?

Reply:
I honestly don't know Google Ads well enough to say, but to be safe I would assume that it could be a duplicate. Now, how to implement it safest? Is it enough to

```sql
row_number() over (partition by gclid, source_relation order by date_day) as rn,
```

Or do we need to safeguard from possible duplicates in any additional way?

Reply:
Could find no docs on the topic, so I sent a question to Google Ads Support, let's see what they'll say.

Reply:
Got a response from Google!

>The Google Click ID (GCLID) is a unique parameter passed in the URL with ad clicks that identifies the campaign and other click attributes associated with the ad for ad tracking and campaign attribution.
As a result, you can safely use the "GCLID" as a unique identifier as it is not repeated between the various accounts.

I think we can close this question then :)